tests/e2e/watch_delay_test.go

```diff
@@ -246,3 +253,241 @@ func continuouslyExecuteGetAll(ctx context.Context, t *testing.T, g *errgroup.Gr
 		return nil
 	})
 }
+
+// TestWatchOnStreamMultiplex ensures slow etcd watchers throws terminal ErrCompacted error if its
+// next batch of events to be sent are compacted.
+func TestWatchOnStreamMultiplex(t *testing.T) {
```

Comment:
Good job on the test, I have confirmed that it reproduces the problem nicely. 

Is there any way to simplify the code?

Reply:
Working on it.

Reply:
Done. Let me know if this is not simplified enough.